src/DotVVM.Framework/Compilation/DefaultViewCompiler.cs

```diff
@@ -104,62 +99,13 @@ protected virtual CSharpCompilation AddToCompilation(CSharpCompilation compilati
-                    .Select(a => GetAssemblyCache().GetAssemblyMetadata(a.Key).WithAliases(ImmutableArray.Create(a.Value, "global"))));
-        }
-
-        private CompiledAssemblyCache GetAssemblyCache()
-        {
-            return assemblyCache;
-        }
-
-        private IEnumerable<Assembly> BuildReferencedAssembliesCache()
-        {
-            var diAssembly = typeof(ServiceCollection).Assembly;
-
-            var references = diAssembly.GetReferencedAssemblies().Select(Assembly.Load)
-                .Concat(markupConfiguration.Assemblies.Select(e => Assembly.Load(new AssemblyName(e))))
-                .Concat(new[] {
-                    diAssembly,
-                    Assembly.Load(new AssemblyName("mscorlib")),
-                    Assembly.Load(new AssemblyName("System.ValueTuple")),
-                    typeof(IServiceProvider).Assembly,
-                    typeof(RuntimeBinderException).Assembly,
-                    typeof(DynamicAttribute).Assembly,
-                    typeof(DotvvmConfiguration).Assembly,
-#if DotNetCore
-                    Assembly.Load(new AssemblyName("System.Runtime")),
-                    Assembly.Load(new AssemblyName("System.Collections.Concurrent")),
-                    Assembly.Load(new AssemblyName("System.Collections")),
-#else
-                    typeof(List<>).Assembly
-#endif
-                });
-
-            var netstandardAssembly = AppDomain.CurrentDomain.GetAssemblies().FirstOrDefault(a => a.GetName().Name == "netstandard");
-            if (netstandardAssembly != null)
-            {
-                references = references.Concat(new[] { netstandardAssembly });
-            }
-            else
-            {
-                try
-                {
-                    // netstandard assembly is required for netstandard 2.0 and in some cases
-                    // for netframework461 and newer. netstandard is not included in netframework452
-                    // and will throw FileNotFoundException. Instead of detecting current netframework
-                    // version, the exception is swallowed.
-                    references = references.Concat(new[] { Assembly.Load(new AssemblyName("netstandard")) });
-                }
-                catch (FileNotFoundException) { }
-            }
-
-            return references.Distinct().ToList();
+                    .Select(a => assemblyCache.GetAssemblyMetadata(a.Key).WithAliases(ImmutableArray.Create(a.Value, "global"))));
         }
-
+        
         public virtual CSharpCompilation CreateCompilation(string assemblyName)
         {
             return CSharpCompilation.Create(assemblyName, options: new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary))
-                .AddReferences(referencedAssembliesCache.Value.Select(a => assemblyCache.GetAssemblyMetadata(a)));
+                .AddReferences(assemblyCache.GetAllAssemblies().Select(a => assemblyCache.GetAssemblyMetadata(a)));
```

Comment:
This used to reference only some assemblies and now we reference everything when the experimental option is disabled, right? I don't know if that will break, but I'd guess there is a reson why the logic used to be so complex.

Reply:
Good catch - I've distinguished between the referenced assemblies and "all" assemblies, and work only with referenced assemblies here.